test/integration/smtpEmail.test.ts

```diff
@@ -0,0 +1,146 @@
+import { expect } from 'chai'
+import express from 'express'
+import { afterEach, beforeEach, describe, it } from 'mocha'
+
+import http from 'http'
+import { container } from 'tsyringe'
+import { z } from 'zod'
+import { Env } from '../../src/env.js'
+import { resetContainer } from '../../src/ioc.js'
+import createHttpServer from '../../src/server.js'
+import VeritableCloudagentEvents from '../../src/services/veritableCloudagentEvents.js'
+import { cleanupCloudagent } from '../helpers/cloudagent.js'
+import { withCompanyHouseMock } from '../helpers/companyHouse.js'
+import { cleanup } from '../helpers/db.js'
+import { validCompanyNumber } from '../helpers/fixtures.js'
+import { post } from '../helpers/routeHelper.js'
+import { delay } from '../helpers/util.js'
+
+const ToSchema = z.array(z.string())
+
+const EmailItemSchema = z.object({
+  isRelayed: z.boolean(),
+  deliveredTo: z.string().email(),
+  id: z.string().uuid(),
+  from: z.string().email(),
+  to: ToSchema,
+  receivedDate: z.string().datetime(),
+  subject: z.string(),
+  attachmentCount: z.number().int(),
+  isUnread: z.boolean(),
+})
+const EmailResponseSchema = z.object({
+  results: z.array(EmailItemSchema),
+})
+
+describe('SMTP email', () => {
+  let server: { app: express.Express; cloudagentEvents: VeritableCloudagentEvents }
+
+  withCompanyHouseMock()
+
+  describe('create invitation and check it has been registered in the email server (happy path)', function () {
+    beforeEach(async () => {
+      process.env.EMAIL_TRANSPORT = 'SMTP_EMAIL'
+      resetContainer()
+      container.registerInstance(Env, new Env())
```

Comment:
I really don't like this hack (which I advised you to create). There's a couple of things I'd like you to try:

1. if you remove `container.registerInstance(Env, new Env())` do the tests still pass?
2. now rename the string of the top level describe of any other test file in the integration tests to start with a letter bigger than `S`. I want to change the order some tests run to make sure the cleanup we're doing here actually works
3. If both of these two are fine move the logic of overriding the env and reseting the contianer to a helper that executes the `beforeEach` and `afterEach`. That I think will be clean enough.
4. If either 1 or 2 don't pass we should probably jump on a call